rollup/da_syncer/da/calldata_blob_source.go

```diff
@@ -105,58 +128,139 @@ func (ds *CalldataBlobSource) processRollupEventsToDA(rollupEvents l1.RollupEven
 			if !ok {
 				return nil, fmt.Errorf("unexpected type of rollup event: %T", rollupEvent)
 			}
-			if entry, err = ds.getCommitBatchDA(commitEvent); err != nil {
-				return nil, fmt.Errorf("failed to get commit batch da: %v, err: %w", rollupEvent.BatchIndex().Uint64(), err)
+
+			// if this is a different commit transaction, we need to create a new DA
+			if lastCommitTransactionHash != commitEvent.TxHash() && len(lastCommitEvents) > 0 {
```

Comment:
this is a bit unintuitive at first glance, so do other places of calling `getAndAppendCommitBatchDA`. but I cannot think of a better way, too. it's due to multiple commit batch events belonging to a same transaction (with multiple blobs).

Reply:
Yeah it is a bit counterintuitive. I added some more comments to make it more clear in 53b6ebf8b0f8d0ba0ccc100a1c9f063becb5e0c2